### PR TITLE
TouchIcon revisions

### PR DESCRIPTION
**Pull request by linc**
Addendum to #207

### CODE REVIEW

File: plugins/TouchIcon/class.touchicon.plugin.php [hunk 1]
```diff
@@ -95,7 +96,7 @@ public function SettingsController_TouchIcon_Create($Sender) {
     * @return string Path to icon
     */
    public function getIconPath() {
-      $Icon = C('Garden.TouchIcon') ? Gdn_Upload::Url(C('Garden.TouchIcon')) : self::DEFAULT_PATH;
+      $Icon = C('Garden.TouchIcon') ? Gdn_Upload::Url(C('Garden.TouchIcon')) : Url(self::DEFAULT_PATH);
```
Comment (tburry) [inline plugins/TouchIcon/class.touchicon.plugin.php]: This should be `Asset()`, not `Url()`.

File: plugins/TouchIcon/class.touchicon.plugin.php [hunk 2]
```diff
@@ -95,7 +96,7 @@ public function SettingsController_TouchIcon_Create($Sender) {
     * @return string Path to icon
     */
    public function getIconPath() {
```
Comment (tburry) [inline plugins/TouchIcon/class.touchicon.plugin.php]: If this returns a URL and not a path then maybe the method should be renamed.

File: plugins/TouchIcon/class.touchicon.plugin.php [hunk 3]
```diff
@@ -45,7 +45,7 @@ public function OnDisable() {
    public function Structure() {
       // Backwards compatibility with v1.
       if (C('Plugins.TouchIcon.Uploaded')) {
-         SaveToConfig('Garden.TouchIcon', PATH_ROOT.'/uploads/TouchIcon/apple-touch-icon.png');
+         SaveToConfig('Garden.TouchIcon', 'TouchIcon/apple-touch-icon.png');
```
Comment (tburry) [inline plugins/TouchIcon/class.touchicon.plugin.php]: This is wrong. It should be blank for the default path or else you will be calling `Gdn_Upload::Url()` on a path that isn't in the uploads folder.

Comment (linc) [inline plugins/TouchIcon/class.touchicon.plugin.php]: No, the 1.0 version always saved the user-uploaded icon as this name in the uploads folder. The default is a constant.

The `Plugins.TouchIcon.Uploaded` config is no longer used; we're using it here to detect preexisting installs.

**Comment by linc** [comment]
Updated

**Pull request merged**